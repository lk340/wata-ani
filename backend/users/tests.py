from django.urls import reverse
from rest_framework import status

from _helpers.test_set_up import TestSetUp


class UserListTestCase(TestSetUp):
    user_list_endpoint = reverse("api-user-list")

    def test_get_users_unauthenticated(self):
        self.client.credentials(HTTP_AUTHORIZATION="Bearer blahBlahToken")
        response = self.client.get(self.user_list_endpoint)
        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)

    def test_get_users_authenticated(self):
        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer {self.token}")
        response = self.client.get(self.user_list_endpoint)
        self.assertEqual(response.status_code, status.HTTP_200_OK)


class UserDetailTestCase(TestSetUp):
    def generate_user_detail_endpoint(self):
        self.user_detail_endpoint = reverse("api-user-detail", kwargs={"pk": self.user.id})

    def test_get_user_unauthenticated(self):
        self.generate_user_detail_endpoint()
        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer blahBlahToken")
        response = self.client.get(self.user_detail_endpoint)
        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)

    def test_get_user_authenticated(self):
        self.generate_user_detail_endpoint()
        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer {self.token}")
        response = self.client.get(self.user_detail_endpoint)
        self.assertEqual(response.status_code, status.HTTP_200_OK)

    def test_patch_user_unauthenticated(self):
        self.generate_user_detail_endpoint()
        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer blahBlahToken")

        response = self.client.get(self.user_detail_endpoint)
        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)

        data = {"username": "foobarUser", "email": "foobar@user.com"}
        response = self.client.patch(self.user_detail_endpoint, data)
        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)

    def test_patch_user_authenticated(self):
        self.generate_user_detail_endpoint()
        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer {self.token}")

        response = self.client.get(self.user_detail_endpoint)
        self.assertEqual(response.status_code, status.HTTP_200_OK)
        self.assertEqual(response.data["username"], "testUser")
        self.assertEqual(response.data["email"], "test@user.com")

        data = {"username": "foobarUser", "email": "foobar@user.com"}
        response = self.client.patch(self.user_detail_endpoint, data)
        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
        self.assertEqual(response.data["username"], "foobarUser")
        self.assertEqual(response.data["email"], "foobar@user.com")

    def test_delete_user_unauthenticated(self):
        self.generate_user_detail_endpoint()
        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer blahBlahToken")
        response = self.client.delete(self.user_detail_endpoint)
        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)

    def test_delete_user_authenticated(self):
        self.generate_user_detail_endpoint()
        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer {self.token}")
        response = self.client.delete(self.user_detail_endpoint)
        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
        self.assertEqual(response.data["detail"], "Your account has been successfully deleted.")
