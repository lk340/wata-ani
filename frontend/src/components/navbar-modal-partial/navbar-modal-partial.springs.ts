import * as Spring from "react-spring";

import * as Context from "@/context";
import * as Constants from "@/utils/style/constants";

export function container(mode: Context.Theme.Mode) {
	return Spring.useSpring({
		to:
			mode === "light"
				? {
						color: Constants.theme.text.light,
						backgroundColor:
							Constants.theme.components.navbarModalPartial.background.light,
						border: `${Constants.theme.components.navbarModalPartial.border.light} solid 1px`,
				  }
				: {
						color: Constants.theme.text.dark,
						backgroundColor:
							Constants.theme.components.navbarModalPartial.background.dark,
						border: `${Constants.theme.components.navbarModalPartial.border.dark} solid 1px`,
				  },
		config: { duration: 100 },
	});
}

export function header(mode: Context.Theme.Mode) {
	return Spring.useSpring({
		to:
			mode === "light"
				? {
						backgroundColor: Constants.theme.components.navbarModalPartial.header.light,
						borderBottom: `${Constants.theme.components.navbarModalPartial.border.light} solid 1px`,
				  }
				: {
						backgroundColor: Constants.theme.components.navbarModalPartial.header.dark,
						borderBottom: `${Constants.theme.components.navbarModalPartial.border.dark} solid 1px`,
				  },
		config: { duration: 100 },
	});
}
