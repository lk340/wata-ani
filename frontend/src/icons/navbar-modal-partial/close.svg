<svg xmlns="http://www.w3.org/2000/svg" width="10" height="10" viewBox="0 0 10 10" fill="none"><g clip-path="url(#A)" fill="#6ba6ff"><path d="M.3 1.71l7.994 7.994a1 1 0 0 0 1.414-1.414L1.713.295a1 1 0 0 0-1.414 0 1 1 0 0 0 0 1.414z"/><path d="M8.293.295L.3 8.29a1 1 0 1 0 1.414 1.414L9.707 1.71A1 1 0 1 0 8.293.295z"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h10v10H0z"/></clipPath></defs></svg>