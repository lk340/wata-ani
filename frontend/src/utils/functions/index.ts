// =============================== //
// ↓↓↓ Data Structures Utility ↓↓↓ //
// =============================== //

export function removeArrayElement<ArrayElement>(
	array: ArrayElement[],
	index: number,
): ArrayElement[] {
	array.splice(index, 1);
	return array;
}

export function capitalizeFirstLetter(wordOrPhrase: string): string {
	const firstLetter = wordOrPhrase[0].toUpperCase();
	const restOfWordOrPhrase = wordOrPhrase.slice(1);
	return firstLetter + restOfWordOrPhrase;
}
