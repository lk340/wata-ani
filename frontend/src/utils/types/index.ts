export * from "./_form";
export * from "./_redux";
export * from "./_styled";
