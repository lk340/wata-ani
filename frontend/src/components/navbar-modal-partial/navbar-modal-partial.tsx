import * as React from "react";

import * as Context from "@/context";
import * as Animations from "@/utils/style/animations";

import * as Styled from "./navbar-modal-partial.styled";
import * as Springs from "./navbar-modal-partial.springs";

export const NavbarModalPartial = () => {
	const { theme } = Context.Theme.useThemeContext();

	const animateContainer = Springs.container(theme.state.mode);

	return (
		<Styled.NavbarModalPartial style={animateContainer}>
			NavbarModalPartial
		</Styled.NavbarModalPartial>
	);
};
