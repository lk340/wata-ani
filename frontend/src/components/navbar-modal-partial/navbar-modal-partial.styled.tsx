import styled, { css, createGlobalStyle } from "styled-components";
import { animated } from "react-spring";

import * as Constants from "@/utils/style/constants";
import * as Snippets from "@/utils/style/snippets";

import close from "@/icons/navbar-modal-partial/close.svg";

// ============================ //
// ↓↓↓ Navbar Modal Partial ↓↓↓ //
// ============================ //

export const NavbarModalPartial = styled(animated.div)`
	${Snippets.absolute("80px", "0px")};
	border-radius: ${Constants.borderRadius.components.navbarModalPartial.container};
	overflow: hidden;
`;

// ============== //
// ↓↓↓ Header ↓↓↓ //
// ============== //

export const NavbarModalPartialHeader = styled(animated.div)`
	${Snippets.flex("row", "space-between", "center")};
	height: ${Constants.size.components.navbarModalPartial.header.height}px;
	padding: 0px ${Constants.sidePaddings.mobile}px;
`;

export const NavbarModalPartialHeaderText = styled(animated.span)`
	display: block;
	font-size: ${Constants.fontSizes.components.navbarModalPartial.header};
	font-weight: bold;
`;

export const NavbarModalPartialHeaderClose = styled("img").attrs(() => ({
	src: close,
	alt: "navbar modal partial close icon",
}))`
	${Snippets.square(`${Constants.size.components.navbarModalPartial.header.close}px`)};
	cursor: pointer;
`;

// ============ //
// ↓↓↓ Body ↓↓↓ //
// ============ //

export const NavbarModalPartialBody = styled(animated.div)`
	padding: ${Constants.sidePaddings.mobile}px;
`;
