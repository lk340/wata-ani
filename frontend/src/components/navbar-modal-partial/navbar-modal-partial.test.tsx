import * as React from "react";
import * as Testing from "@testing-library/react";

import * as Tests from "@/utils/tests";

import { NavbarModalPartial } from "./navbar-modal-partial";

describe("NavbarModalPartial Component", () => {
	it("matches the snapshot", () => {
		const { asFragment } = Tests.renderWithContext(<NavbarModalPartial />);

		expect(asFragment(<NavbarModalPartial />)).toMatchSnapshot();
	});
});
