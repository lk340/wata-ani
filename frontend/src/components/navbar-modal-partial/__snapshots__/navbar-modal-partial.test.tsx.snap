// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarModalPartial Component matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  top: 80px;
  right: 0px;
  bottom: auto;
  left: auto;
  z-index: 1;
  border-radius: 0.3125rem;
  overflow: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 70px;
  padding: 0px 30px;
}

.c2 {
  display: block;
  font-size: 1rem;
  font-weight: bold;
}

.c3 {
  width: 20px;
  height: 20px;
  cursor: pointer;
}

.c4 {
  padding: 30px;
}

<div
    class="c0"
    style="background-color: rgb(249, 249, 249); border: 1px solid rgba(219, 217, 224, 1); box-shadow: 0px 3px 8px 0px rgba(189, 190, 198, 0.5019607843137255);"
  >
    <div
      class="c1"
      style="background-color: rgb(255, 255, 255); border-bottom: 1px solid rgba(219, 217, 224, 1);"
    >
      <span
        class="c2"
        style="color: rgb(113, 117, 138);"
      />
      <img
        alt="navbar modal partial close icon"
        class="c3"
        src="[object Object]"
      />
    </div>
    <div
      class="c4"
      style="background-color: rgb(249, 249, 249);"
    />
  </div>
</DocumentFragment>
`;
